Index: b/allauth/headless/account/views.py
===================================================================
--- a/allauth/headless/account/views.py
+++ b/allauth/headless/account/views.py
@@ -60,6 +60,7 @@
 from allauth.headless.internal import authkit
 from allauth.headless.internal.restkit.response import ErrorResponse
 
+from zango.core.utils import get_auth_priority
 
 class RequestLoginCodeView(APIView):
     input_class = RequestLoginCodeInput
@@ -297,7 +298,9 @@
         if r429:
             return r429
         self.input.save(request)
-        if account_settings.PASSWORD_RESET_BY_CODE_ENABLED:
+        password_policy = get_auth_priority(policy='password_policy', request=request)
+        reset_policy = password_policy.get('reset', {})
+        if not reset_policy.get('enabled', False):
             return AuthenticationResponse(request)
         return response.RequestPasswordResponse(request)
 
@@ -313,7 +316,9 @@
 
     def handle(self, request, *args, **kwargs):
         self.process = None
-        if account_settings.PASSWORD_RESET_BY_CODE_ENABLED:
+        password_policy = get_auth_priority(policy="password_policy", request=request)
+        password_reset_policy = password_policy.get("reset", {})
+        if password_reset_policy.get("by_code", False):
             self.process = (
                 password_reset_by_code.PasswordResetVerificationProcess.resume(
                     self.request
@@ -387,7 +392,7 @@
         if request.user.is_authenticated:
             self.user = request.user
         elif request.method != "POST":
-            return AuthenticationResponse(request)
+            return response.AuthenticationResponse(request)
         else:
             self.verification_stage_process = EmailVerificationProcess.resume(request)
             if (
@@ -463,7 +468,7 @@
         if request.user.is_authenticated:
             self.user = request.user
         elif request.method == "GET":
-            return AuthenticationResponse(request)
+            return response.AuthenticationResponse(request)
         elif request.method == "POST":
             stage = LoginStageController.enter(request, PhoneVerificationStage.key)
             if stage:
