Index: b/allauth/headless/internal/sessionkit.py
===================================================================
--- a/allauth/headless/internal/sessionkit.py
+++ b/allauth/headless/internal/sessionkit.py
@@ -18,8 +18,9 @@
 
 
 def expose_session_token(request):
-    if request.allauth.headless.client != Client.APP:
-        return
+    if getattr(request.allauth, "headless", None):
+        if request.allauth.headless.client != Client.APP:
+            return
     strategy = app_settings.TOKEN_STRATEGY
     hdr_token = strategy.get_session_token(request)
     modified = request.session.modified
