Index: b/allauth/headless/adapter.py
===================================================================
--- a/allauth/headless/adapter.py
+++ b/allauth/headless/adapter.py
@@ -74,7 +74,8 @@
         if account_settings.USER_MODEL_USERNAME_FIELD:
             kwargs["username"] = user_username(user)
         if user.pk:
-            email = EmailAddress.objects.get_primary_email(user)
+            email = None
+            # email = EmailAddress.objects.get_primary_email(user)
         else:
             email = None
         kwargs.update(
