Index: b/allauth/headless/base/response.py
===================================================================
--- a/allauth/headless/base/response.py
+++ b/allauth/headless/base/response.py
@@ -11,7 +11,8 @@
 from allauth.headless.constants import Flow
 from allauth.headless.internal import authkit
 from allauth.headless.internal.restkit.response import APIResponse
-from allauth.mfa import app_settings as mfa_settings
+
+from zango.core.utils import get_auth_priority
 
 
 class BaseAuthenticationResponse(APIResponse):
@@ -21,11 +22,15 @@
             adapter = get_adapter()
             data["user"] = adapter.serialize_user(user)
             data["methods"] = get_authentication_records(request)
+            redirect_url = get_auth_priority(config_key="redirect_url", request=request, user=user)
+            data["redirect_url"] = redirect_url
             status = status or HTTPStatus.OK
         else:
             status = status or HTTPStatus.UNAUTHORIZED
         if status != HTTPStatus.OK:
-            data["flows"] = self._get_flows(request, user)
+            flows = self._get_flows(request, user)
+            if flows:
+                data["next_step"] = flows
         meta = {
             "is_authenticated": user and user.is_authenticated,
         }
@@ -38,26 +43,26 @@
 
     def _get_flows(self, request, user):
         auth_status = authkit.AuthenticationStatus(request)
-        ret = []
-        if user and user.is_authenticated:
-            ret.extend(flows.reauthentication.get_reauthentication_flows(user))
-        else:
-            if not allauth_settings.SOCIALACCOUNT_ONLY:
-                ret.append({"id": Flow.LOGIN})
-            if account_settings.LOGIN_BY_CODE_ENABLED:
-                ret.append({"id": Flow.LOGIN_BY_CODE})
-            if (
-                get_account_adapter().is_open_for_signup(request)
-                and not allauth_settings.SOCIALACCOUNT_ONLY
-            ):
-                ret.append({"id": Flow.SIGNUP})
-            if allauth_settings.SOCIALACCOUNT_ENABLED:
-                from allauth.headless.socialaccount.response import provider_flows
-
-                ret.extend(provider_flows(request))
-            if allauth_settings.MFA_ENABLED:
-                if mfa_settings.PASSKEY_LOGIN_ENABLED:
-                    ret.append({"id": Flow.MFA_LOGIN_WEBAUTHN})
+        # ret = []
+        # if user and user.is_authenticated:
+        #     ret.extend(flows.reauthentication.get_reauthentication_flows(user))
+        # else:
+        #     if not allauth_settings.SOCIALACCOUNT_ONLY:
+        #         ret.append({"id": Flow.LOGIN})
+        #     if account_settings.LOGIN_BY_CODE_ENABLED:
+        #         ret.append({"id": Flow.LOGIN_BY_CODE})
+        #     if (
+        #         get_account_adapter().is_open_for_signup(request)
+        #         and not allauth_settings.SOCIALACCOUNT_ONLY
+        #     ):
+        #         ret.append({"id": Flow.SIGNUP})
+        #     if allauth_settings.SOCIALACCOUNT_ENABLED:
+        #         from allauth.headless.socialaccount.response import provider_flows
+
+        #         ret.extend(provider_flows(request))
+        #     if allauth_settings.MFA_ENABLED:
+        #         if mfa_settings.PASSKEY_LOGIN_ENABLED:
+        #             ret.append({"id": Flow.MFA_LOGIN_WEBAUTHN})
         stage_key = None
         stage = auth_status.get_pending_stage()
         if stage:
@@ -68,27 +73,34 @@
                 stage_key = lsk
         if stage_key:
             pending_flow = {"id": stage_key, "is_pending": True}
+            if stage and stage_key == "role_selection":
+                self._enrich_role_selection_flow(stage, pending_flow)
+                return pending_flow
+            if stage and stage_key == "set_password":
+                self._enrich_set_password_flow(stage, pending_flow, request)
+                return pending_flow
             if stage and stage_key == Flow.MFA_AUTHENTICATE:
-                self._enrich_mfa_flow(stage, pending_flow)
-            self._upsert_pending_flow(ret, pending_flow)
-
-        if (
-            not allauth_settings.SOCIALACCOUNT_ONLY
-            and account_settings.PASSWORD_RESET_BY_CODE_ENABLED
-        ):
-            from allauth.account.internal.flows import password_reset_by_code
-
-            ret.append(
-                {
-                    "id": Flow.PASSWORD_RESET_BY_CODE,
-                    "is_pending": bool(
-                        password_reset_by_code.PasswordResetVerificationProcess.resume(
-                            request
-                        )
-                    ),
-                }
-            )
-        return ret
+                self._enrich_mfa_flow(stage, pending_flow, request)
+                return pending_flow
+            # self._upsert_pending_flow(ret, pending_flow)
+
+        # if (
+        #     not allauth_settings.SOCIALACCOUNT_ONLY
+        #     and account_settings.PASSWORD_RESET_BY_CODE_ENABLED
+        # ):
+        #     from allauth.account.internal.flows import password_reset_by_code
+
+        #     ret.append(
+        #         {
+        #             "id": Flow.PASSWORD_RESET_BY_CODE,
+        #             "is_pending": bool(
+        #                 password_reset_by_code.PasswordResetVerificationProcess.resume(
+        #                     request
+        #                 )
+        #             ),
+        #         }
+        #     )
+        return {}
 
     def _upsert_pending_flow(self, flows, pending_flow):
         flow = next((flow for flow in flows if flow["id"] == pending_flow["id"]), None)
@@ -97,16 +109,32 @@
         else:
             flows.append(pending_flow)
 
-    def _enrich_mfa_flow(self, stage, flow: dict) -> None:
+    def _enrich_mfa_flow(self, stage, flow: dict, request) -> None:
+        print("user is ", stage.login.user)
+        flow["metadata"] = {}
         from allauth.mfa.adapter import get_adapter as get_mfa_adapter
-        from allauth.mfa.models import Authenticator
 
         adapter = get_mfa_adapter()
-        types = []
-        for typ in Authenticator.Type:
-            if adapter.is_mfa_enabled(stage.login.user, types=[typ]):
-                types.append(typ)
-        flow["types"] = types
+        enabled, types = adapter.is_mfa_enabled(stage.login.user)
+        if enabled:
+            if len(request.session.get("account_authentication_methods", [])) > 0:
+                latest_auth_method = request.session["account_authentication_methods"][0]
+                if latest_auth_method.get("email"):
+                    flow["metadata"]["type"] = "sms"
+                else:
+                    flow["metadata"]["type"] = "email"
+
+    def _enrich_role_selection_flow(self, stage, flow: dict) -> None:
+        flow["metadata"] = {}
+        roles = {}
+        for role in stage.login.user.roles.all():
+            roles[role.pk] = role.name
+        flow["metadata"]["roles"] = roles
+
+    def _enrich_set_password_flow(self, stage, flow: dict, request) -> None:
+        flow["metadata"] = {}
+        policy = get_auth_priority(policy="password_policy", request=request, user=stage.login.user)
+        flow["metadata"]["password_policy"] = policy
 
 
 class AuthenticationResponse(BaseAuthenticationResponse):
